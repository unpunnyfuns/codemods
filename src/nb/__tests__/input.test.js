import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/input', () => {
  it('migrates Input component', () => {
    const output = testTransform('nb/input', 'nb/input', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Input',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
