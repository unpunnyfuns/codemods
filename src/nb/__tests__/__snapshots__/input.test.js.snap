// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/input > migrates Input component 1`] = `
"import { Input } from 'native-base'

export function Example() {
  return (
    <>
      {/* Basic Input */}
      <Input placeholder="Enter name" value={name} onChangeText={setName} />

      {/* Input with keyboard type */}
      <Input
        placeholder="Email address"
        value={email}
        onChangeText={setEmail}
        keyboardType="email-address"
      />

      {/* Input with testID */}
      <Input
        placeholder="Password"
        value={password}
        onChangeText={setPassword}
        testID="password-input"
      />

      {/* Input with layout props */}
      <Input placeholder="Search" value={search} onChangeText={setSearch} mt="md" mb="sm" />

      {/* Input with complex props (will warn) */}
      <Input
        placeholder="Phone"
        value={phone}
        onChangeText={setPhone}
        InputLeftElement={<Icon name="Phone" />}
      />
    </>
  )
}
"
`;
